src/gui/search/searchsortmodel.cpp

```diff
@@ -29,17 +29,89 @@
 #include "searchsortmodel.h"
 
 SearchSortModel::SearchSortModel(QObject *parent)
-    : QSortFilterProxyModel(parent)
+    : base(parent)
+    , m_isNameFilterEnabled(false)
+    , m_minSeeds(0)
+    , m_maxSeeds(-1)
+    , m_minLeechs(0)
+    , m_maxLeechs(-1)
+    , m_minSize(0)
+    , m_maxSize(-1)
 {
 }
 
+void SearchSortModel::enableNameFilter(bool enable)
+{
+    m_isNameFilterEnabled = enable;
+}
+
+void SearchSortModel::setNameFilter(const QString &searchTerm)
+{
+    m_searchTerm = searchTerm;
+    if (searchTerm.length() > 2
+        && searchTerm.at(0) == QLatin1Char('"') && searchTerm.at(searchTerm.length() - 1) == QLatin1Char('"')) {
+        m_searchTermWords.append(m_searchTerm.mid(1, m_searchTerm.length() - 2));
```

Comment:
What a nice bug! It exists, but does not expose itself. Wish they all learn from this one.